### PR TITLE
[SYCL][Fusion][Test] XFAIL `KernelFusion/Reduction/reduction.cpp`

### PR DESCRIPTION
**Pull request by victor-eds**
XFAIL failing test on CPU until the issue is fixed.

### CODE REVIEW

**Review by sommerlukas — APPROVED** [review]

File: sycl/test-e2e/KernelFusion/Reduction/reduction.cpp [hunk 1]
```diff
@@ -1,5 +1,6 @@
 // RUN: %{build} -fsycl-embed-ir -o %t.out
 // RUN: %{run} %t.out
+// XFAIL: cpu
```
Comment (aelovikov-intel) [inline sycl/test-e2e/KernelFusion/Reduction/reduction.cpp:3]: Interesting... If pre-commit passes, that would mean that the test is skipped there.

Comment (victor-eds) [reply sycl/test-e2e/KernelFusion/Reduction/reduction.cpp:3]: If this is caused by a race condition depending on the strategy being used for the reduction, we may only run into the error in some devices

**Comment by aelovikov-intel** [comment]
Failures in post-commit (for future references/searches):

```
# Arc GPU:
Failed Tests (3):
  SYCL :: ESIMD/lsc/local_accessor_atomic_smoke_cmpxchg.cpp
  SYCL :: ESIMD/lsc/lsc_slm_atomic_smoke.cpp
  SYCL :: USM/usm_pooling.cpp

```

First two are likely due to https://github.com/intel/llvm/pull/12535#issuecomment-1917602350.

The other one is probably just flaky, failed like this:
```
# RUN: at line 27
env UR_L0_DEBUG=1 SYCL_PI_LEVEL_ZERO_USM_RESIDENT=0 SYCL_PI_LEVEL_ZERO_USM_ALLOCATOR=";;2M,4,4M" env ONEAPI_DEVICE_SELECTOR=level_zero:gpu  /__w/llvm/llvm/build-e2e/USM/Output/usm_pooling.cpp.tmp.out h 2>&1 | /__w/llvm/llvm/toolchain/bin/FileCheck /__w/llvm/llvm/llvm/sycl/test-e2e/USM/usm_pooling.cpp --check-prefix CHECK-15
# executed command: env UR_L0_DEBUG=1 SYCL_PI_LEVEL_ZERO_USM_RESIDENT=0 'SYCL_PI_LEVEL_ZERO_USM_ALLOCATOR=;;2M,4,4M' env ONEAPI_DEVICE_SELECTOR=level_zero:gpu /__w/llvm/llvm/build-e2e/USM/Output/usm_pooling.cpp.tmp.out h
# note: command had no output on stdout or stderr
# executed command: /__w/llvm/llvm/toolchain/bin/FileCheck /__w/llvm/llvm/llvm/sycl/test-e2e/USM/usm_pooling.cpp --check-prefix CHECK-15
# note: command had no output on stdout or stderr
# RUN: at line 28
env UR_L0_DEBUG=1 SYCL_PI_LEVEL_ZERO_USM_RESIDENT=0 SYCL_PI_LEVEL_ZERO_USM_ALLOCATOR=";;2M,4,4M" env ONEAPI_DEVICE_SELECTOR=level_zero:gpu  /__w/llvm/llvm/build-e2e/USM/Output/usm_pooling.cpp.tmp.out d 2>&1 | /__w/llvm/llvm/toolchain/bin/FileCheck /__w/llvm/llvm/llvm/sycl/test-e2e/USM/usm_pooling.cpp --check-prefix CHECK-15
# executed command: env UR_L0_DEBUG=1 SYCL_PI_LEVEL_ZERO_USM_RESIDENT=0 'SYCL_PI_LEVEL_ZERO_USM_ALLOCATOR=;;2M,4,4M' env ONEAPI_DEVICE_SELECTOR=level_zero:gpu /__w/llvm/llvm/build-e2e/USM/Output/usm_pooling.cpp.tmp.out d
# note: command had no output on stdout or stderr
# error: command failed with exit status: -6
# executed command: /__w/llvm/llvm/toolchain/bin/FileCheck /__w/llvm/llvm/llvm/sycl/test-e2e/USM/usm_pooling.cpp --check-prefix CHECK-15
# note: command had no output on stdout or stderr
```

**Comment by aelovikov-intel** [comment]
@victor-eds , we've enabled OpenCL CPU testing on another runner (with Intel CPU this time) and the test seems to pass consistently there, see my comments summarizing Nigthly results over the weekend in https://github.com/intel/llvm/pull/12586.

**Pull request merged**